sql/pgtap.sql.in

```diff
@@ -11518,3 +11577,376 @@ RETURNS TEXT AS $$
+                            --and the column is not part of PK when we want to leave PK as is
+                            t.attname::text != all((select _keys(_table.table_schema, _table.table_name, 'p'))::text[]) and
+                            _leave_primary_key
+                        ) or
+                        --we can drop a NOT NULL constraint as there is no PK already
+                        not _leave_primary_key
+                    );
+
+                _fake_ddl = _fake_ddl || _not_null_ddl || ';';
+            else
+                _fake_ddl = null;
+            end if;
+
+            if _fake_ddl is not null then
+                execute _fake_ddl;
+            end if;
+
+            if _drop_partitions then
+                if pg_version_num() < 100000 then
+                    raise exception 'Sorry, but declarative partitioning was introduced only starting with PostgreSQL version 10.';
+                end if;
+                for _part_table in select _parts from _parts(_table.table_schema, _table.table_name) loop
+                    _fake_ddl = 'drop table if exists ' || _part_table._parts || ';';
+                    execute _fake_ddl;
+                end loop;
+            end if;
+        end loop;
+end $function$;
+
+create or replace function call_count(
+	_call_count int
+	, _func_schema name
+	, _func_name name
+	, _func_args name[])
+ RETURNS text
+ LANGUAGE plpgsql
+AS $function$
+declare
+    _actual_call_count int;
+	_track_functions_setting text;
+begin
+	select current_setting('track_functions') into _track_functions_setting;
+
+	if _track_functions_setting != 'all' then
+	    return fail('track_functions setting is not set. Must be all');
+	end if;
+
+	select calls into _actual_call_count
+	from pg_stat_xact_user_functions
+	where funcid = _get_func_oid(_func_schema, _func_name, _func_args);
+
+    return ok(
+        _actual_call_count = _call_count
+        , format('routine %I.%I must has been called %L times, actual call count is %L'
+        	, _func_schema, _func_name, _call_count, _actual_call_count)
+    );
+end $function$;
+
+create or replace function drop_prepared_statement(_statements text[])
+returns setof bool as $$
+declare
+    _statement record;
+begin
+	for _statement in select _name from unnest(_statements) as t(_name) loop
+	    if exists(select * from pg_prepared_statements where "name" = _statement._name) then
+	        EXECUTE format('deallocate %I;', _statement._name);
+	        return next true;
+		else
+			return next false;
+	    end if;
+	end loop;
+end
+$$
+language plpgsql;
+
+
+create or replace function print_table_as_json(in _table_schema text, in _table_name text)
+returns void
+ language plpgsql
+AS $function$
+declare
+    _ddl text;
+    _json text;
+    _columns text;
+--returns a query which you can execute and see your table as normal dataset
+--you can find the returned query in the output window in DBeaver, where we see raise notice command output
```

Comment:
I don't understand what Beaver has to do with this. I have never used it, and it shouldn't be a requirement for users to have it.